{"ast":null,"code":"import axios from 'axios';\nimport { useContext } from 'react';\nconst baseUrl = 'http://localhost:5000/api/users';\n\nconst getAll = async () => {\n  const config = {\n    method: 'get',\n    url: baseUrl,\n    headers: {}\n  };\n\n  try {\n    let res = await axios(config);\n    return res.data;\n  } catch (error) {\n    alert(error);\n    return;\n  }\n\n  ;\n};\n/*const getProfessionalsResults = async (person) => {\r\n    const config = {\r\n        method: 'get',\r\n        url: baseUrl + '/professionals/',\r\n        headers: {}\r\n    }\r\n};*/\n\n\nconst signup = async (newObject, setError) => {\n  await axios.post(`${baseUrl}/register/`, newObject).then(response => {\n    return response;\n  }).catch(error => {\n    if (error.response) {\n      let customError = error.response;\n      customError.userMessage = \"Rekisteröityminen epäonnistui\";\n      setError(customError);\n      return error;\n    }\n  });\n};\n\nconst login = async (newObject, setError, login, setUID) => {\n  await axios.post(`${baseUrl}/login/`, newObject).then(response => {\n    login();\n    setUID(newObject.uid);\n    return response;\n  }).catch(error => {\n    if (error.response) {\n      let customError = error.response;\n      customError.userMessage = \"Käyttäjää ei ole olemassa. Tarkista käyttäjätunnuksesi ja yritä uudelleen!\";\n      setError(customError);\n      return error;\n    }\n  });\n};","map":{"version":3,"sources":["C:/Users/jenni/Documents/GitHub/harjoitus4-JenniMERN/frontend/src/user/services/user-services.js"],"names":["axios","useContext","baseUrl","getAll","config","method","url","headers","res","data","error","alert","signup","newObject","setError","post","then","response","catch","customError","userMessage","login","setUID","uid"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAAQC,UAAR,QAAyB,OAAzB;AAEA,MAAMC,OAAO,GAAG,iCAAhB;;AAEA,MAAMC,MAAM,GAAG,YAAY;AACvB,QAAMC,MAAM,GAAG;AACXC,IAAAA,MAAM,EAAE,KADG;AAEXC,IAAAA,GAAG,EAAEJ,OAFM;AAGXK,IAAAA,OAAO,EAAE;AAHE,GAAf;;AAKA,MAAI;AACA,QAAIC,GAAG,GAAG,MAAMR,KAAK,CAACI,MAAD,CAArB;AACA,WAAOI,GAAG,CAACC,IAAX;AACH,GAHD,CAGE,OAAOC,KAAP,EAAa;AACXC,IAAAA,KAAK,CAACD,KAAD,CAAL;AACA;AACH;;AAAA;AACJ,CAbD;AAeA;AACA;AACA;AACA;AACA;AACA;AACA;;;AACA,MAAME,MAAM,GAAG,OAAOC,SAAP,EAAkBC,QAAlB,KAA8B;AACzC,QAAMd,KAAK,CAACe,IAAN,CAAY,GAAEb,OAAQ,YAAtB,EAAmCW,SAAnC,EACLG,IADK,CACAC,QAAQ,IAAI;AACd,WAAOA,QAAP;AACH,GAHK,EAILC,KAJK,CAIER,KAAK,IAAI;AACb,QAAGA,KAAK,CAACO,QAAT,EAAkB;AACd,UAAIE,WAAW,GAAGT,KAAK,CAACO,QAAxB;AACAE,MAAAA,WAAW,CAACC,WAAZ,GAA0B,+BAA1B;AACAN,MAAAA,QAAQ,CAACK,WAAD,CAAR;AACA,aAAOT,KAAP;AACH;AACJ,GAXK,CAAN;AAYH,CAbD;;AAeA,MAAMW,KAAK,GAAG,OAAOR,SAAP,EAAkBC,QAAlB,EAA4BO,KAA5B,EAAmCC,MAAnC,KAA8C;AACxD,QAAMtB,KAAK,CAACe,IAAN,CAAa,GAAEb,OAAQ,SAAvB,EAAiCW,SAAjC,EACLG,IADK,CACAC,QAAQ,IAAG;AACbI,IAAAA,KAAK;AACLC,IAAAA,MAAM,CAACT,SAAS,CAACU,GAAX,CAAN;AACA,WAAON,QAAP;AACH,GALK,EAMLC,KANK,CAMER,KAAK,IAAI;AACb,QAAGA,KAAK,CAACO,QAAT,EAAkB;AACd,UAAIE,WAAW,GAAGT,KAAK,CAACO,QAAxB;AACAE,MAAAA,WAAW,CAACC,WAAZ,GAA0B,4EAA1B;AACAN,MAAAA,QAAQ,CAACK,WAAD,CAAR;AACA,aAAOT,KAAP;AACH;AACJ,GAbK,CAAN;AAcH,CAfD","sourcesContent":["import axios from 'axios';\r\nimport {useContext} from 'react';\r\n\r\nconst baseUrl = 'http://localhost:5000/api/users'\r\n\r\nconst getAll = async () => {\r\n    const config = {\r\n        method: 'get',\r\n        url: baseUrl,\r\n        headers: {}\r\n    }\r\n    try {\r\n        let res = await axios(config);\r\n        return res.data;\r\n    } catch (error){\r\n        alert(error);\r\n        return;\r\n    };\r\n};\r\n\r\n/*const getProfessionalsResults = async (person) => {\r\n    const config = {\r\n        method: 'get',\r\n        url: baseUrl + '/professionals/',\r\n        headers: {}\r\n    }\r\n};*/\r\nconst signup = async (newObject, setError) =>{\r\n    await axios.post(`${baseUrl}/register/`, newObject)\r\n    .then(response => {\r\n        return response;\r\n    })\r\n    .catch (error => {\r\n        if(error.response){\r\n            let customError = error.response;\r\n            customError.userMessage = \"Rekisteröityminen epäonnistui\";\r\n            setError(customError);\r\n            return error;\r\n        }\r\n    })\r\n}\r\n\r\nconst login = async (newObject, setError, login, setUID) => {\r\n    await axios.post (`${baseUrl}/login/`, newObject)\r\n    .then(response =>{\r\n        login();\r\n        setUID(newObject.uid);\r\n        return response;\r\n    })\r\n    .catch (error => {\r\n        if(error.response){\r\n            let customError = error.response;\r\n            customError.userMessage = \"Käyttäjää ei ole olemassa. Tarkista käyttäjätunnuksesi ja yritä uudelleen!\";\r\n            setError(customError);\r\n            return error;\r\n        }\r\n    })\r\n}"]},"metadata":{},"sourceType":"module"}